src/doc/unstable-book/src/language-features/proc-macro.md

```diff
@@ -6,5 +6,153 @@ The tracking issue for this feature is: [#38356]
+
+#### Importing
+
+As referenced above, the new procedural macros are not meant to be imported via `#[macro_use]` and will throw an 
+error if they are. Instead, they are meant to be imported like any other item in Rust, with `use` statements:
+
+```rust
+// Where `my_proc_macros` is some crate of type `proc_macro`
+extern crate my_proc_macros;
+
+// And declares a `#[proc_macro] pub fn my_bang_macro()` at its root.
+use my_proc_macros::my_bang_macro;
+
+fn main() {
+    println!("{}", my_bang_macro!());
+}
+```
+
+#### Error Reporting
+
+Currently the only support for error reporting from procedural macros is through panics. Any panics in a procedural 
+macro implementation will be caught by the compiler and turned into an error message pointing to the problematic 
+invocation.
+
+#### `TokenStream`
+
+The `proc_macro::TokenStream` type is hardcoded into the signatures of procedural macro functions. It is a wrapper 
+around the compiler's internal representation for a given chunk of Rust code. It currently provides only two APIs:
+
+* A `FromStr` impl that attempts to parse the source string as Rust code; it can take anything from a single expression 
+up to any number of items (function, type or `impl` declarations), as long as the string is valid Rust syntax. If 
+the syntax is valid but not in the context in which the macro is invoked, an error is thrown outside the macro.
+
+* A `Display` impl which writes out the contained tokens. Of course, the blanket `ToString` impl applies 
+to this type as well.
+
+### Function-like / Bang-style Procedural Macros
+
+These are procedural macros that are invoked like `macro_rules!` macros. They are declared as public functions in crates 
+of the `proc_macro` type and using the `#[proc_macro]` attribute. The name of the declared function becomes the name of 
+the macro as it is to be imported and used. The function must be of the kind `fn(TokenStream) -> TokenStream` where the 
+sole argument is the input to the macro and the return type is the macro's output.
+
+This kind of macro can expand to anything that is valid for the context it is invoked in, including expressions and
+statements, as well as items.
+
+**Note**: invocations of this kind of macro require a wrapping `[]`, `{}` or `()` like regular macros, but these do not 
+appear in the input, only the tokens between them. The tokens between the braces do not need to be valid Rust syntax.
+
+`my_macro_crate/lib.rs`:
+```rust
+#![feature(proc_macro)]
+
+// This is always necessary to get the `TokenStream` typedef.
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+
+#[proc_macro]
+pub fn say_hello(_input: TokenStream) -> TokenStream {
+    // This macro will accept any input because it ignores it. 
+    // To enforce correctness in macros which don't take input,
+    // you may want to add `assert!(_input.to_string().is_empty());`.
+    "println!(\"Hello, world!\")".parse().unwrap()
+}
+```
+
+`my_macro_user/main.rs`:
```

Comment:
Do I need to show adding the dependency on `my_macro_crate` to `my_macro_user/Cargo.toml`?